admin/runner.py

```diff
@@ -27,21 +48,21 @@ class CommandProtocol(LineOnlyReceiver, object):
         If the command finished successfully, will fire with ``None``.
         Otherwise, errbacks with the reason.
     :ivar file-like output: For logging.
+
+    :ivar defaultdict _fds: Mapping from fds to `_LineParsers`.
     """
-    delimiter = b'\n'
+    def __init__(self):
+        self._fds = defaultdict(lambda: _LineParser(output=self.output))
 
-    def connectionMade(self):
-        self.transport.disconnecting = False
+    def childDataReceived(self, childFD, data):
+        self._fds[childFD].dataReceived(data)
 
-    def connectionLost(self, reason):
-        if reason.check(ConnectionDone):
+    def processEnded(self, reason):
```

Comment:
Whichever it ends up as can you comment please - I have read the definitions of both and am not clear on why it would be either.